wfl/Profiles/FoodOrderCollectionProfile.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Hypermedia.Affordances;
+using Hypermedia.Transforms;
+using Hypermedia;
+using Symphono.Wfl.Models;
+using Symphono.Wfl.Controllers;
+using System;
+
+namespace Symphono.Wfl.Profiles
+{
+    public class FoodOrderCollectionProfile : EntityProfile<FoodOrderCollection>
+    {
+        protected override void Configure(EntityConfiguration<FoodOrderCollection> configuration)
+        {
+            configuration
+                .UseRepresentationTransform(representation => representation
+                    .WithRepresentation("food-order-collection")
+                    .WithRepresentation("collection")
+                )
+                .UseEnumerableEmbeddedSubEntityTransform(
+                    c => c.FoodOrders,
+                    embedded => embedded
+                        .WithRelation("item")
+                        .WithConfiguration(embeddedConfiguration => embeddedConfiguration
+                            .UseProfile<FoodOrder, FoodOrderProfile>()
+                        )
+                )
+                .WithLink<FoodOrderCollection, FoodOrdersController>(
+                    "self",
+                    c => fc => fc.GetAsync(c.Criteria)
+                )
+                .UseActionTransform(actions => actions
```

Comment:
How will the user know what values can be used for `Status` in the filter? It may be a good idea to create a route to return the allowed values from the enumeration. That way someone can hit a link to get a list of all the statuses that can be used here. That will need a hypermedia profile as well.